homework-submission/todo-app/src/component/week3Components/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
+import Name from "./name";
+import Img from "./img";
+import Region from "./region";
+import Gender from "./gender";
+import "core-js";
+
+function Week3Components() {
+  const [data, setData] = useState("");
+  const [id, setId] = useState(-1);
+
+  useEffect(() => {
+    const fetchData = async url => {
+      await fetch(url)
+        .then(response => response.json())
+        .then(content => setData(content));
+    };
+    try {
+      fetchData("https://uinames.com/api/?amount=10");
+    } catch (err) {
+      console.log(err);
+    }
+
+    return () => fetchData("https://uinames.com/api/?amount=10");
+  }, []);
+
+  document.body.onkeyup = e => {
+    if (e.keyCode === 32 && id <= 8) {
+      setId(id + 1);
```

Comment:
Not so 'random' in your case 😉

Reply:
HeHe, it was 'random' in a word 'ordered random'.hhh... but fixed in week4.